src/corefx/System.Globalization.Native/localeStringData.cpp

```diff
@@ -240,3 +255,83 @@ extern "C" int32_t GetLocaleInfoString(const UChar* localeName, LocaleStringData
 
 	return UErrorCodeToBool(status);
 }
+
+/*
+Function:
+NormalizeTimePattern
+
+Convert an ICU non-localized time pattern to .NET format
+*/
+void NormalizeTimePattern(const UnicodeString *srcPattern, UnicodeString *destPattern)
+{
+	// An srcPattern example: "h:mm:ss a"
+	// A destPattern example: "h:mm:ss tt"
+	destPattern->remove();
+
+	for (int i = 0; i <= srcPattern->length() - 1; i++)
+	{
+		UChar ch = srcPattern->charAt(i);
+		switch (ch)
+		{
+		case ':':
+		case '.':
```

Comment:
Note in theory there could be other separators besides : and . but all existing locales in ICU use those two. An alternative implementation would let in more characters here